df_py/predictoor/models.py

```diff
@@ -71,14 +70,72 @@ def accuracy(self):
         return self._accuracy
 
 
+class PredictionSummary:
+    @enforce_types
+    def __init__(self, prediction_count, correct_prediction_count, contract_addr):
+        self.prediction_count = prediction_count
+        self.correct_prediction_count = correct_prediction_count
+        self.contract_addr = contract_addr
+
+    @property
+    def accuracy(self) -> float:
+        if self.prediction_count == 0:
+            return 0
+        return self.correct_prediction_count / self.prediction_count
+
+
 class Predictoor(PredictoorBase):
     @enforce_types
     def __init__(self, address: str):
         super().__init__(address, 0, 0, 0)
         self._predictions: List[Prediction] = []
 
+    def get_prediction_summary(self, contract_addr: str) -> PredictionSummary:
+        """
+        Get the prediction summary for a specific contract address.
+
+        @param contract_addr: The contract address for which to get the prediction summary.
+
+        @return:
+            PredictionSummary - The prediction summary for the specified contract address.
+        """
+        prediction_count = 0
+        correct_prediction_count = 0
+        for prediction in self._predictions:
+            if prediction.contract_addr == contract_addr:
+                prediction_count += 1
+                if prediction.is_correct:
```

Comment:
I think you can try skipping on the negative, to make this more readable.
e.g. if prediction.contract_addr != contract_addr: continue
then you don't have to indent the rest, same for the other condition

it's one of my favourite tricks :) https://kaolin.tech/2016/10/untangling-conditionals-with-if-inversion/